packages/core/src/radio-button/RadioButton.tsx

```diff
@@ -90,6 +91,24 @@ export const RadioButton = forwardRef<HTMLLabelElement, RadioButtonProps>(
 
     const radioGroup = useRadioGroup();
 
+    const {
+      "aria-describedby": inputDescribedBy,
+      "aria-labelledby": inputLabelledBy,
+      ...restInputProps
+    } = inputProps;
+
+    const {
+      a11yProps: {
+        "aria-describedby": formFieldDescribedBy,
+        "aria-labelledby": formFieldLabelledBy,
+      } = {},
+      disabled: formFieldDisabled,
+      validationStatus: formFieldValidationStatus,
+    } = useFormFieldProps();
+
+    const disabled = formFieldDisabled ?? disabledProp;
+    const error = formFieldValidationStatus === "error" ?? errorProp;
```

Comment:
i think this should be done on the RadioButtonGroup. it's unlikely the radio button will be used directly inside the FF

Reply:
updated